api/src/main/java/org/openmrs/validator/PersonNameValidator.java

```diff
@@ -164,4 +180,29 @@ private String getFieldKey(String field, boolean arrayInd, boolean testInd) {
 		return testInd ? field : arrayInd ? "names[0]." + field : "name." + field;
 	}
 	
-}
+	
+	private boolean startsOrEndsWithWhitespace(String value) {
+		boolean startsOrEndsWithWhitespace = false;
+		if (StringUtils.isNotBlank(value)) {
+		    int lastIndex = value.length() - 1;
+		    if (Character.isSpaceChar(value.charAt(0))) {
+			    startsOrEndsWithWhitespace = true;
+		    } else if (Character.isSpaceChar(value.charAt(lastIndex))) {
+		    	
+			    startsOrEndsWithWhitespace = true; 
+		    }
+		} else {
+			startsOrEndsWithWhitespace = false;
+		}
+		return startsOrEndsWithWhitespace;		
+	}
+	
+	private boolean matchNamePattern(String value, String namePattern) {
+		boolean match = true;
+		if (value != null && StringUtils.isNotBlank(value)) {
```

Comment:
Doesn't StringUtils.isNotBlank already compare for null?

Reply:
@dkayiwa  I think, the `StingUtils.isNotBlank` has the capability to check for `null` also. 
```
 StringUtils.isNotBlank(null)      = false
 StringUtils.isNotBlank("")        = false
 StringUtils.isNotBlank(" ")       = false
```